packages/ec-core/src/operator/recombinator/mod.rs

```diff
@@ -3,25 +3,272 @@ use rand::rngs::ThreadRng;
+///
+///     fn recombine(
+///         &self,
+///         (mut first_parent, second_parent): (Vec<T>, Vec<T>),
+///         rng: &mut ThreadRng,
+///     ) -> anyhow::Result<Vec<T>> {
+///         let index = rng.gen_range(0..first_parent.len());
+///         first_parent[index] = second_parent[index];
+///         Ok(first_parent)
+///     }
+/// }
+///
+/// // A simple `Operator` that takes a `Vec<bool>` and returns the number
+/// // of `true` values in the genome.
+/// struct CountTrue;
+///
+/// impl Operator<Vec<bool>> for CountTrue {
+///     type Output = usize;
+///     type Error = anyhow::Error;
+///
+///     fn apply(&self, genome: Vec<bool>, _: &mut ThreadRng) -> Result<Self::Output, Self::Error> {
+///         Ok(genome.iter().filter(|&&x| x).count())
+///     }
+/// }
+/// impl Composable for CountTrue {}
+///
+/// // If we swap in exactly one value from the `second_parent` we
+/// // should have 3 `true` values in the resulting genome.
+/// let first_parent = vec![true, true, true, true];
+/// let second_parent = vec![false, false, false, false];
+///
+/// let recombinator = SwapOne;
+/// let recombine = Recombine::new(recombinator);
+/// let count_true = CountTrue;
+/// let chain = recombine.then(count_true);
+///
+/// let count = chain
+///     .apply((first_parent, second_parent), &mut thread_rng())
+///     .unwrap();
+/// assert_eq!(count, 3);
+/// ```
+///
+/// We can also pass a _reference_ to a [`Recombinator`] (i.e.,
+/// `&Recombinator`) to a [`Recombine`], allowing us to pass references to
+/// recombinators into chains of operators without requiring or giving up
+/// ownership of the recombinator.
+///
+/// ```
+/// # use rand::{rngs::ThreadRng, thread_rng, Rng};
+/// #
+/// # use ec_core::operator::recombinator::{Recombinator, Recombine};
+/// # use ec_core::operator::{Composable, Operator};
+/// #
+/// # // A simple `Recombinator` that swaps one element from the second parent
+/// # // into the corresponding position in the first parent.
+/// # struct SwapOne;
+/// #
+/// # impl<T: Copy> Recombinator<(Vec<T>, Vec<T>)> for SwapOne {
+/// #     type Output = Vec<T>;
+/// #
+/// #     fn recombine(
+/// #         &self,
+/// #         (mut first_parent, second_parent): (Vec<T>, Vec<T>),
+/// #         rng: &mut ThreadRng,
+/// #     ) -> anyhow::Result<Vec<T>> {
+/// #         let index = rng.gen_range(0..first_parent.len());
+/// #         first_parent[index] = second_parent[index];
+/// #         Ok(first_parent)
+/// #     }
+/// # }
+/// #
+/// # // A simple `Operator` that takes a `Vec<bool>` and returns the number
+/// # // of `true` values in the genome.
+/// # struct CountTrue;
+/// #
+/// # impl Operator<Vec<bool>> for CountTrue {
+/// #     type Output = usize;
+/// #     type Error = anyhow::Error;
+/// #
+/// #     fn apply(&self, genome: Vec<bool>, _: &mut ThreadRng) ->
+/// # Result<Self::Output, Self::Error> {         Ok(genome.iter().filter(|&&x|
+/// # x).count())     }
+/// # }
```

Comment:
something went south with the formatting here

Reply:
Yeah, `rustfmt` has lost its mind here. I can try to "fix" it, and it just auto-reformats to this nonsense... :-(

Reply:
Oooh – it doesn't anymore. Maybe they fixed a bug in the `rustfmt` update!